registry/regclient/profile.go

```diff
@@ -135,3 +170,48 @@ func (c Client) doJSONProfileReq(method string, p *registry.Profile) (*registry.
 	env.Data.Peername = env.Data.Username
 	return env.Data, nil
 }
+
+// RegistryResponse is a generic container for registry requests
+// TODO(dustmop): Only used currently for ProveKey
+type RegistryResponse struct {
+	Data map[string]string
+	Meta struct {
+		Error  string
+		Status string
+		Code   int
+	}
+}
+
+// doJSONProfileReq sends a json body to the registry
```

Comment:
This can potentially also be moved to the HTTPClient.
I'd be happy if this merged soon and I can refactor accordingly.